providers/supermicro/supermicro.go

```diff
@@ -184,6 +188,33 @@ func (c *Client) Open(ctx context.Context) (err error) {
 	return nil
 }
 
+// PowerStateGet gets the power state of a BMC machine
+func (c *Client) PowerStateGet(ctx context.Context) (state string, err error) {
+	if c.serviceClient == nil || c.serviceClient.redfish == nil {
+		return "", errors.Wrap(bmclibErrs.ErrLoginFailed, "client not initialized")
+	}
+
+	return c.serviceClient.redfish.SystemPowerStatus(ctx)
+}
+
+// PowerSet sets the power state of a server
+func (c *Client) PowerSet(ctx context.Context, state string) (ok bool, err error) {
+	if c.serviceClient == nil || c.serviceClient.redfish == nil {
+		return false, errors.Wrap(bmclibErrs.ErrLoginFailed, "client not initialized")
+	}
+
+	return c.serviceClient.redfish.PowerSet(ctx, state)
+}
```

Comment:
Should the `state` for these methods be a constant with a specific type, something like  `PowerState`?

Reply:
yep thats the plan, although since this will affect _all_ providers and it changes the interface, it needs to be done separately